src/main/java/mate/academy/spring/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -57,4 +58,71 @@ public Optional<MovieSession> get(Long id) {
             throw new DataProcessingException("Can't get a movie session by id: " + id, e);
         }
     }
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movie session " + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public MovieSession update(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.update(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't update movie session " + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public void remove(Long id) {
+        MovieSession movieSession = get(id)
+                .orElseThrow(() -> new RuntimeException("Session with id=" + id + " not found"));
+        Transaction transaction = null;
```

Comment:
Maybe better to pass not Id but movieSession object
and move get logic to service layer
```suggestion
    public void remove(movieSession) {        
        Transaction transaction = null;
```